File: app/src/main/java/xyz/ummo/user/ui/fragments/delegatedService/DelegatedServiceViewModel.java
package xyz.ummo.user.ui.fragments.delegatedService;

import android.app.Application;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import xyz.ummo.user.data.entity.DelegatedServiceEntity;
import xyz.ummo.user.data.repo.AppRepository;

public class DelegatedServiceViewModel extends AndroidViewModel {
    private AppRepository appRepository;
    private LiveData<DelegatedServiceEntity> delegatedServiceEntityLiveData;
    private static final String TAG = "DelegatedServiceVM";

    public DelegatedServiceViewModel(@NonNull Application application) {
        super(application);

        appRepository = new AppRepository(application);
      //  delegatedServiceEntityLiveData = appRepository.getDelegatedServiceEntity();
    }

    public void insertDelegatedService(DelegatedServiceEntity delegatedServiceEntity){
        appRepository.insertDelegatedService(delegatedServiceEntity);
        Log.e(TAG, "insertDelegatedService: SERVICE->"+delegatedServiceEntity.getServiceId());
    }

    public void updateDelegatedService(DelegatedServiceEntity delegatedServiceEntity){
        appRepository.updateDelegatedService(delegatedServiceEntity);
    }

    public LiveData<DelegatedServiceEntity> getDelegatedServiceEntityLiveData(){
        return appRepository.getDelegatedServiceEntityLiveData();
    }
    public void deleteAllDelegatedServices(){
        appRepository.deleteAllDelegatedServices();
    }

    public LiveData<DelegatedServiceEntity> getDelegatedServiceById(String id){
        Log.e(TAG, "getDelegatedServiceById: "+id);
        return appRepository.getDelegatedServiceById(id);
    }

    public LiveData<DelegatedServiceEntity> getDelegatedServiceByProductId(String productId){
        return appRepository.getDelegatedServiceByProductId(productId);
    }
}

File: app/src/main/java/xyz/ummo/user/ui/fragments/delegatedService/DelegatedServiceViewModel.kt
package xyz.ummo.user.ui.fragments.delegatedService

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import timber.log.Timber
import xyz.ummo.user.data.entity.DelegatedServiceEntity
import xyz.ummo.user.data.repo.AppRepository

class DelegatedServiceViewModel(application: Application) : AndroidViewModel(application) {
    private val appRepository: AppRepository = AppRepository(application)
    val delegatedServiceEntityLiveData: LiveData<DelegatedServiceEntity>?
        get() = appRepository.delegatedServiceEntityLiveData

    fun insertDelegatedService(delegatedServiceEntity: DelegatedServiceEntity) {
        appRepository.insertDelegatedService(delegatedServiceEntity)
        Timber.e("insertDelegatedService: SERVICE->%s", delegatedServiceEntity.serviceId)
    }

    fun updateDelegatedService(delegatedServiceEntity: DelegatedServiceEntity?) {
        appRepository.updateDelegatedService(delegatedServiceEntity)
    }

    fun deleteAllDelegatedServices() {
        appRepository.deleteAllDelegatedServices()
    }

    fun getDelegatedServiceById(id: String?): LiveData<DelegatedServiceEntity> {
        Timber.e("getDelegatedServiceById: %s", id)
        return appRepository.getDelegatedServiceById(id)
    }

    fun getDelegatedServiceByProductId(productId: String?): LiveData<DelegatedServiceEntity> {
        return appRepository.getDelegatedServiceByProductId(productId)
    }

    companion object {
        private const val TAG = "DelegatedServiceVM"
    }

    init {
        //  delegatedServiceEntityLiveData = appRepository.getDelegatedServiceEntity();
    }
}